### PR TITLE
Prevent test courses from being overwritten

### PR DESCRIPTION
**Pull request by shanbady**
### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/7373

### Description (What does it do?)
We have a mechanism for tagging certain courses as being in "test_mode" so that they have their contentfiles ingested into qdrant and opensearch but remain "unpublished". There is an issue where scheduled etl backpopulate tasks overwrite this status and remove contentfiles from the indexes.

### How can this be tested?
1. checkout main
2. make sure you have some courses populated. in this case we will use mitxonline which has a course intended to be a test course ```python manage.py backpopulate_mitxonline_data```
3. see that the test_mode on the learning resource is false. also note the id
```python
LearningResource.objects.get(readable_id='course-v1:UAI_SOURCE+UAI.2').test_mode
print(LearningResource.objects.get(readable_id='course-v1:UAI_SOURCE+UAI.2').id)
```
4. use the id to mark it as test_mode when fetching contentfiles:
```python
python manage.py backpopulate_mitxonline_files --resource-ids <resource_id> --test-ids <resource_id> --overwrite
```
5. note that there are not contentfiles for the course run:
```python
LearningResource.objects.get(readable_id='course-v1:UAI_SOURCE+UAI.2').runs.first().content_files.all()
```
6. manually set the run for that course to published:
```python
run = LearningResource.objects.get(readable_id='course-v1:UAI_SOURCE+UAI.2').runs.first()
run.published = True
run.save()
``` 
and re-run ` backpopulate_mitxonline_files --resource-ids <resource_id> --test-ids <resource_id> --overwrite`

7. re-run `python manage.py backpopulate_mitxonline_data` -> note that the contentfiles for that particular run are gone/removed.
8. re-run above steps on this branch and note that contentfiles for test_mode=True resources are persisted. 
9. You should also be able to manually set test_mode back to false and re-run the pipeline to see them get removed again
10. validate that test_mode=True resources do not show up in the search index http://open.odl.local:8063/api/v1/learning_resources_search/?id=<test resource id>

### Checklist:
- [ ] Validate this behaves as expected for test_mode=True and test_mode=False course
- [ ] go through the changes and make sure there are no unintended side effects or edge cases this could raise

### CODE REVIEW

File: learning_resources/etl/loaders.py [hunk 1]
```diff
@@ -329,7 +320,9 @@ def load_run(
             run_id=run_id,
             defaults={**run_data},
         )
-
+        if learning_resource_run.learning_resource.test_mode:
+            learning_resource_run.published = True
+            learning_resource_run.save()
```
[the remark this answers is not in the document]
Comment (shanbady) [reply learning_resources/etl/loaders.py]: I'm not aware of any consequences of setting them all to published if the course is in test mode and the resource is unpublished. I figure we can patch that when/if the need arises.
